### ISSUE TITLE
Get audio data in real-time

### ISSUE THREAD

**Issue by t41372** [ricky0123/vad#122]
Is there a way to get the audio data while the speech is active before it ends? I want to get the audio data when the speech starts, stream it to the back end in real-time and stop streaming when it ends. It seems like the `onFrameProcessed` callback only has a probability property.
Thanks

**Comment by rahulbansal16** [comment]
Which model will you use in the backend to transcribe it? Does the word error rate increase in that way?

**Comment by ricky0123** [comment]
This is something I would like to add better support for in the future, but it will be a significant addition and I'm still not sure what form I want the API to take. In the mean time, have you tried writing separate code to stream audio to your servers and just using the VAD callbacks for voice activity signals?

**Comment by Oudwins** [comment]
@ricky0123 yes I tried to implement streaming using the VAD callbacks with an algorithm that I think could work quite well for the library:
- keep track of the padding frames
- onSpeechStart keep track of speech frames
- onSpeech exceeding min frames flush padding and saved speech frames to server and start streaming all new frames to server
- onSpeechEnd stop streaming new frames to server
- onVADmissfire empty buffered speech frames

But while implementing this I found the issue described in #186
